ocs_ci/deployment/ibm_deployment.py

```diff
@@ -0,0 +1,1024 @@
+    # redefinition is not necessary in normal cases.
+    DEFAULT_STORAGECLASS_LSO = 'localblock'
+
+    def __init__(self):
+        self.platform = config.ENV_DATA['platform']
+        self.ocp_deployment_type = config.ENV_DATA['deployment_type']
+        self.cluster_path = config.ENV_DATA['cluster_path']
+        self.namespace = config.ENV_DATA["cluster_namespace"]
+
+    class OCPDeployment(BaseOCPDeployment):
+        """
+        This class has to be implemented in child class and should overload
+        methods for platform specific config.
+        """
+        pass
+
+    def deploy_cluster(self, log_cli_level='DEBUG'):
+        """
+        We are handling both OCP and OCS deployment here based on flags
+
+        Args:
+            log_cli_level (str): log level for installer (default: DEBUG)
+        """
+        if not config.ENV_DATA['skip_ocp_deployment']:
+            if is_cluster_running(self.cluster_path):
+                logger.warning(
+                    "OCP cluster is already running, skipping installation"
+                )
+            else:
+                try:
+                    self.deploy_ocp(log_cli_level)
+                    self.post_ocp_deploy()
+                except Exception as e:
+                    logger.error(e)
+                    if config.REPORTING['gather_on_deploy_failure']:
+                        collect_ocs_logs('deployment', ocs=False)
+                    raise
+
+        if not config.ENV_DATA['skip_ocs_deployment']:
+            try:
+                self.deploy_ocs()
+            except Exception as e:
+                logger.error(e)
+                if config.REPORTING['gather_on_deploy_failure']:
+                    # Let's do the collections separately to guard against one
+                    # of them failing
+                    collect_ocs_logs('deployment', ocs=False)
+                    collect_ocs_logs('deployment', ocp=False)
+                raise
+        else:
+            logger.warning("OCS deployment will be skipped")
+
+    def deploy_ocp(self, log_cli_level='DEBUG'):
+        """
+        Base deployment steps, the rest should be implemented in the child
+        class.
+
+        Args:
+            log_cli_level (str): log level for installer (default: DEBUG)
+        """
+        self.ocp_deployment = self.OCPDeployment()
+        self.ocp_deployment.deploy_prereq()
+        self.ocp_deployment.deploy(log_cli_level)
+        # logging the cluster UUID so that we can ask for it's telemetry data
+        cluster_id = run_cmd("oc get clusterversion version -o jsonpath='{.spec.clusterID}'")
+        logger.info(f"clusterID (UUID): {cluster_id}")
+
+    def post_ocp_deploy(self):
+        """
+        Function does post OCP deployment stuff we need to do.
+        """
+        set_selinux_permissions()
+        set_registry_to_managed_state()
+        add_stage_cert()
+
+    def label_and_taint_nodes(self):
```

Comment:
We definitely don't want to be duplicating what is already established within the base class. That defeats the purpose of inheriting from that class in the first place and makes it much more difficult to determine what is specifically done for the IBM deployment vs. other deployments. It appears that the entire deployment module has been copy/pasted here and then potentially changed for IBM specific functionality. It would be better to start fresh with a new class that inherits from `Deployment` and only add functionality / override functions as necessary.

Reply:
I trimmed ibm_deployment.py, renamed to ibm.py and removed redundant code.